WordPress/src/main/java/org/wordpress/android/ui/domains/DomainSuggestionsViewHolder.kt

```diff
@@ -33,4 +39,34 @@ class DomainSuggestionsViewHolder(
             }
         }
     }
+
+    private fun getFormattedCost(suggestion: DomainSuggestionResponse, site: SiteModel) = when {
+        suggestion.is_free -> {
+            suggestion.cost
+        }
+        SiteUtils.onFreePlan(site) -> {
```

Comment:
We probably shouldn't use `SiteUtils.onFreePlan` to determine if the domain is going to be free or not, because it's possible for a user on a paid plan to not have any domain credits left, in which case, they are still able to purchase a domain that is not going to be free.

Assuming I'm not missing anything, we should probably reuse some of the logic in the [`DomainRegistrationHandler`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/domainregistration/DomainRegistrationHandler.kt) class to determine if the user has domain credits or not.

Reply:
Thanks for a quick review @renanferrari.  I'll take look at improving this logic in the next PRs